src/main/java/com/petshopadmin/adapter/input/http/species/SpecieController.java

```diff
@@ -44,6 +44,18 @@ public ResponseHTTP getByID(@PathVariable(name = "id", required = true) Long spe
         return new ResponseHTTP("success to get species by id", new SpecieResponseHTTP(specieDomain), null, LocalDateTime.now());
     }
 
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    @GetMapping(path = "name/{name}")
```

Comment:
Pra esse tipo de situação normalmente o pessoal utiliza o query param. Aí, neste caso, é possível realizar queries com os outros parâmetros.

* https://stackoverflow.com/questions/32201441/how-do-i-retrieve-query-parameters-in-a-spring-boot-controller
* https://www.baeldung.com/spring-request-param